app/views/events/_form.html.haml

```diff
@@ -1,7 +1,9 @@
 = simple_form_for resource do |f|
   .section
-    = render 'form_base', f: f
```

Comment:
The following seems to do the trick:

`= f.time_field :start_at, value: f.object.start_at&.to_s(:time), class: 'timepicker'`

* better use `f.object` where `f` is the current form variable. This reduces the dependency on inherited resources etc. and uses the object that has been used to start the form
* The `&.` syntax does only call the following method if the object is not `nil`
* the `to_formatted_s` is aliased to `to_s`

I also think the `value` assignment is not needed for the date part.